app/lua53/lua.h

```diff
@@ -465,6 +465,8 @@ struct lua_Debug {
   struct CallInfo *i_ci;  /* active function */
 };
 
+int lua_main(void);
```

Comment:
I hummed and harred about where to export `lua_main()`. This is roughly equivalent to `main()` in std `lua.c`.  `lua.h` is the public interface to the Lua API, and this main entry is _not_ part of the API and in general if and module writer were to attempt to call this, it would be a disaster.  This is why I left the entry _hidden_ , and declared is `extern` in the file that requires it.

Reply:
Hmm, on the other hand, letting others declare things `extern` elsewhere is a recipe for subtle breakage if the actual prototype _does_ change (been there, done that 😞). Maybe it should simply be dumped into its own header file that gets included by lua.c to ensure the declaration and definition match?